x-pack/plugins/security_solution/server/endpoint/services/actions/clients/sentinelone/sentinel_one_actions_client.ts

```diff
@@ -195,28 +200,54 @@ export class SentinelOneActionsClient extends ResponseActionsClientImpl {
       },
     });
 
-    return this.fetchActionDetails(actionRequestDoc.EndpointActions.action_id);
-  }
-
-  async release(options: IsolationRouteRequestBody): Promise<ActionDetails> {
-    await this.validateRequest(options);
-    await this.sendAction(SUB_ACTION.RELEASE_HOST, {
-      uuid: options.endpoint_ids[0],
-    });
-
-    const actionRequestDoc = await this.writeActionRequestToEndpointIndex({
+    const createPayload: CreateActionPayload = {
       ...options,
-      command: 'unisolate',
-    });
+      command: commandName,
+      action_id: actionRequestDoc.EndpointActions.action_id,
+      user: { username: this.actionsClientOptions.username },
+    };
 
-    await this.writeActionResponseToEndpointIndex({
-      actionId: actionRequestDoc.EndpointActions.action_id,
-      agentId: actionRequestDoc.agent.id,
-      data: {
-        command: actionRequestDoc.EndpointActions.data.command,
-      },
-    });
+    try {
+      const agentId = options.endpoint_ids[0];
+      await updateCases({
+        casesClient: this.options.casesClient,
+        endpointData: [
+          {
+            host: {
+              hostname: actionRequestDoc.EndpointActions.data.hosts?.[agentId].name as string,
+            },
+            agent: {
+              id: agentId,
+              type: options.agent_type || 'endpoint',
+            },
+          },
+        ],
+        createActionPayload: createPayload,
+      });
```

Comment:
Don't use this function. Instead, the `this.updateCases()` method should be used here (its defined in the base class). 

Also: I have several changes in my PR to this client, including the changes to trigger updates to cases - here: https://github.com/elastic/kibana/pull/174691/files#diff-fd830af91a61270f449e1d43e7f1d07f838bd25cc7c22b397555cdf85dc512cbR217-R228

If you can keep the implementation close to that one in my PR, that would be great.

Reply:
Alright, will do!